code/app/src/main/java/com/example/emojibrite/ProfileImageGenerator.java

```diff
@@ -111,15 +110,13 @@ public void onResponse(@NonNull Call call, @NonNull Response response) throws IO
                     Uri imageUri = Uri.parse(response.request().url().toString());
 
                     // Store the URI in the database
-                    Database database = new Database(context);
+                    Database database = new Database();
```

Comment:
For this, everytime getProfileImage() is called, it creates a new instance of database. I would suggest putting this "Database database = new Database(); in the attributes portion
public class ProfileImageGenerator {
    OkHttpClient client = new OkHttpClient();
    String name;
    **Database database = new Database();**
    String Uid;

Reply:
the thing is profileimage exists just for this method  so it should be fine. there are no  other useful methods other than this in the class to begin with